pkg/controller/apmserverelasticsearchassociation/apmserverelasticsearchassociation_controller.go

```diff
@@ -252,6 +264,23 @@ func (r *ReconcileApmServerElasticsearchAssociation) reconcileInternal(apmServer
 		return commonv1.AssociationFailed, err
 	}
 
+	allowed, err := r.accessReviewer.AccessAllowed(apmServer.Spec.ServiceAccountName, apmServer.Namespace, &es)
+	if err != nil {
+		return commonv1.AssociationPending, err
+	}
+	if !allowed {
+		log.Info("Association not allowed",
+			"as_name", apmServer.Name,
+			"as_namespace", apmServer.Namespace,
+			"serviceAccount", apmServer.Spec.ServiceAccountName,
+			"es_name", es.Name,
+			"es_namespace", es.Namespace,
+		)
+		// Ensure that user in Elasticsearch is deleted to prevent illegitimate access
+		err := user.DeleteUser(r.Client, NewUserLabelSelector(assocKey))
```

Comment:
Should we also call association.RemoveAssociationConf?

Reply:
We should maybe extract that block and unit test it does the right thing when the association is not allowed.